### ISSUE TITLE
Binding to chained command with alias fails

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#3002]
`config.bind(',f', 'nop ;; wq')` in `config.py` fails as it doesn't know the `wq` alias. Without a chained command, it works.

cc @pkillnine

**Comment by pkillnine** [comment]
Note:

Qutebrowser will also recognise the alias as valid if the alias is the first in a chained command (not just without a chained command).

**Comment by The-Compiler** [comment]
This has to do with how the command parser parses aliases - I tried to fix that, but it makes things handling with commands a lot more complicated because now parsing the first command of a chain only you can suddenly end up with multiple commands. I think the better solution would just be to not validate commands while binding them, but just letting them fail afterwards...

Here's a broken patch for the alias fixes:

```diff
diff --git a/qutebrowser/commands/runners.py b/qutebrowser/commands/runners.py
index 9d480cc5d..1abe489f0 100644
--- a/qutebrowser/commands/runners.py
+++ b/qutebrowser/commands/runners.py
@@ -127,7 +127,7 @@ class CommandParser:
             new_cmd += ' '
         return new_cmd
 
-    def _parse_all_gen(self, text, aliases=True, *args, **kwargs):
+    def _parse_all_gen(self, text, *args, **kwargs):
         """Split a command on ;; and parse all parts.
 
         If the first command in the commandline is a non-split one, it only
@@ -135,7 +135,6 @@ class CommandParser:
 
         Args:
             text: Text to parse.
-            aliases: Whether to handle aliases.
             *args/**kwargs: Passed to parse().
 
         Yields:
@@ -145,14 +144,11 @@ class CommandParser:
         if not text:
             raise cmdexc.NoSuchCommandError("No command given")
 
-        if aliases:
-            text = self._get_alias(text, text)
-
         if ';;' in text:
             # Get the first command and check if it doesn't want to have ;;
             # split.
             first = text.split(';;')[0]
-            result = self.parse(first, *args, **kwargs)
+            result = next(self._parse(first, *args, **kwargs))
             if result.cmd.no_cmd_split:
                 sub_texts = [text]
             else:
@@ -160,13 +156,13 @@ class CommandParser:
         else:
             sub_texts = [text]
         for sub in sub_texts:
-            yield self.parse(sub, *args, **kwargs)
+            yield from self._parse(sub, *args, **kwargs)
 
     def parse_all(self, *args, **kwargs):
         """Wrapper over parse_all."""
         return list(self._parse_all_gen(*args, **kwargs))
 
-    def parse(self, text, *, fallback=False, keep=False):
+    def _parse(self, text, *, fallback=False, keep=False, aliases=True):
         """Split the commandline text into command and arguments.
 
         Args:
@@ -175,14 +171,22 @@ class CommandParser:
                       unknown.
             keep: Whether to keep special chars and whitespace
 
-        Return:
-            A ParseResult tuple.
+        Yields:
+            ParseResult tuples. There can be multiple if we find an alias with a
+            split command.
         """
-        cmdstr, sep, argstr = text.partition(' ')
-
-        if not cmdstr and not fallback:
+        if not text.strip() and not fallback:
             raise cmdexc.NoSuchCommandError("No command given")
 
+        if aliases:
+            new_text = self._get_alias(text)
+            if new_text is not None:
+                yield from self.parse_all(text, fallback=fallback, keep=keep,
+                                          aliases=aliases)
+                return
+
+        cmdstr, sep, argstr = text.partition(' ')
+
         if self._partial_match:
             cmdstr = self._completion_match(cmdstr)
 
@@ -193,7 +197,8 @@ class CommandParser:
                 raise cmdexc.NoSuchCommandError(
                     '{}: no such command'.format(cmdstr))
             cmdline = split.split(text, keep=keep)
-            return ParseResult(cmd=None, args=None, cmdline=cmdline)
+            yield ParseResult(cmd=None, args=None, cmdline=cmdline)
+            return
 
         args = self._split_args(cmd, argstr, keep)
         if keep and args:
@@ -203,7 +208,7 @@ class CommandParser:
         else:
             cmdline = [cmdstr] + args[:]
 
-        return ParseResult(cmd=cmd, args=args, cmdline=cmdline)
+        yield ParseResult(cmd=cmd, args=args, cmdline=cmdline)
 
     def _completion_match(self, cmdstr):
         """Replace cmdstr with a matching completion if there's only one match.
diff --git a/qutebrowser/completion/completer.py b/qutebrowser/completion/completer.py
index efd6b3490..7770715b5 100644
--- a/qutebrowser/completion/completer.py
+++ b/qutebrowser/completion/completer.py
@@ -132,7 +132,7 @@ class Completer(QObject):
             # Only ":", empty part under the cursor with nothing before/after
             return [], '', []
         parser = runners.CommandParser()
-        result = parser.parse(text, fallback=True, keep=True)
+        result = parser.parse_all(text, fallback=True, keep=True)[0]
         # pylint: disable=not-an-iterable
         parts = [x for x in result.cmdline if x]
         # pylint: enable=not-an-iterable
```

**Comment by The-Compiler** [comment]
Fixed by 64b783d9c056045bcb5caa3d9ab8543b54fc4e18 (forgot to reference the issue in the commit).
